### DISCUSSION TITLE
How to get abstract syntax tree in hyperformula?

### DISCUSSION THREAD

**Discussion by dlee992** [handsontable/hyperformula#625] [Feature requests]
### Description

I am developing an office addins for Excel, now I want to obtain the abstract syntax tree and dependency graph/tree from a formula extracted from Excel Worksheet.

I searched in your documentation and issues, but found nothing about it.

Do you already provide this API? Or, maybe I can obtain these stuff through a few changes in your code.

**Comment by AMBudnik** [comment]
Hi @dlee992 

Sorry for keeping you waiting. 

At this moment we do not have a public API that could visually reflect the abstract syntax tree. But, we gather all the needed information and are able to retrieve it. I've crafted a demo to help us discuss the topic https://codesandbox.io/s/gallant-dirac-c4to9 

What the demo does is
1. Return an array of all addresses and ranges whose computation depends on input address or range provided (line `23`)
2. Return all addresses and ranges necessary for computation of a given address or range (line `27`)
3. Return a list of cached formulas (line `34`)
4. Return a total list of nodes with a notation of a vertex type (line `37`)


Reclaiming values from cache might not be the best idea if you allow CRUD operations in the project. But if you do then please let me know and I'll adjust the demo.

If there's anything you'd want to be returned besides the listed values I'd be happy to discuss the topic further.

**Comment by sequba** [comment]
Related: https://github.com/handsontable/hyperformula/issues/854
